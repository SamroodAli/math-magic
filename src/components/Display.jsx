import React from 'react';
import PropTypes from 'prop-types';

class Display extends React.PureComponent {
  text = (value, text) => <p>{`${text}: ${value}`}</p>;

  render() {
    const { result } = this.props;
    return <div data-testid="display">{this.text(result, 'Result')}</div>;
  }
}

export default Display;

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};
