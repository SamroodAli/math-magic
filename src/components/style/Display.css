.display {
  height: 100px;
  color: #fff;
  font-weight: 700;
  font-size: 2rem;
  padding: 0 1rem;
  display: flex;
  justify-content: flex-end;
}
