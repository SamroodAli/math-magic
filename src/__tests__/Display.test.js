import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

test('Display panel displays all calculator buttons', () => {
  const tree = renderer.create(<Display result="10" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Display component renders div with test-id display', async () => {
  const display = render(<Display result="10" />);
  const div = await display.findByTestId('display');
  expect(div).toBeDefined();
});

test('Display component shows total on screen', () => {
  render(<Display result="10" />);
  const total = screen.getAllByText('10');
  expect(total).toBeDefined();
});
