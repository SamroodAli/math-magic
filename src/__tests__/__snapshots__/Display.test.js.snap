// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display panel displays all calculator buttons 1`] = `
<div
  data-testid="display"
>
  <p>
    Result: 10
  </p>
</div>
`;
